ast/src/main/java/com/graphicsfuzz/common/ast/decl/ArrayInfo.java

```diff
@@ -17,37 +17,58 @@
 package com.graphicsfuzz.common.ast.decl;
 
 import com.graphicsfuzz.common.ast.IAstNode;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.IntConstantExpr;
 import com.graphicsfuzz.common.ast.visitors.IAstVisitor;
+import com.graphicsfuzz.common.ast.visitors.UnsupportedLanguageFeatureException;
 import java.util.Optional;
 
 public class ArrayInfo implements IAstNode {
 
```

Comment:
Please add a comment to explain the reason for having two size-related fields.

Reply:
Now that I look at this again, I think it would be better to make constantSize be Optional<Integer> (no need for it to actually be IntConstantExpression), initialize it to Optional.empty(), and then only set it during constant folding.  That seems cleaner than having it temporarily be an arbitrary expression.

For the comments, I would say something like:

"originalSize" is the expression in the original shader representing the array size, and is empty if no size was given.  If "originalSize" is present, constant folding can be used to turn the expression into an integer value, which can be stored in "constantSize".  It is useful to keep "originalSize" around to allow the shader to be pretty-printed in its original form.

Reply:
Changed constantSize to integer, rename originalSize to just sizeExpr.